samcli/local/apigw/local_apigw_service.py

```diff
@@ -461,7 +481,9 @@ def _event_headers(flask_request, port):
 
         headers_dict["X-Forwarded-Port"] = str(port)
         multi_value_headers_dict["X-Forwarded-Port"] = [str(port)]
-
+        if cors_headers:
+            headers_dict.update(cors_headers)
+            multi_value_headers_dict.update(cors_headers)
```

Comment:
multi_value_headers value is a list, so this breaks that.

Reply:
I don't think they are lists. They are dictionaries defined in _event_headers() as        
headers_dict = {}
multi_value_headers_dict = {}